Add RW61x monolithic build support on BLE applications

yeaissa opened the pull request:
The feature allows to link the binary blobs with BLE applications during the build, forming
one single monolithic image that can be flashed directly to the board.

axelnxp committed:
mcux: add Kconfig to configure NXP BLE monolithic feature

The monolithic feature allows to link the binary blobs with
the application during the build.

Signed-off-by: Axel Le Bourhis <axel.lebourhis@nxp.com>
Signed-off-by: Yassine El Aissaoui <yassine.elaissaoui@nxp.com>

axelnxp committed:
soc: rw6xx: Enable NXP_BLE_MONOLITHIC

Enable monolithic build for all BLE apps.

Signed-off-by: Axel Le Bourhis <axel.lebourhis@nxp.com>
Signed-off-by: Yassine El Aissaoui <yassine.elaissaoui@nxp.com>

axelnxp committed:
boards: rd_rw612_bga: flash size with monolithic feature

When building with monolithic feature enabled, the firmwares binaries
will be linked during the build with the application. This means we can
remove the application size limit due to the firmware storage in flash
not useful anymore.

Signed-off-by: Axel Le Bourhis <axel.lebourhis@nxp.com>
Signed-off-by: Yassine El Aissaoui <yassine.elaissaoui@nxp.com>

yeaissa committed:
samples: tests: Create dedicated test section for rd_rw612_bga

Monolithic build is enabled by default and it
requires the presence of the binary blobs.
CI will fail since it doesn't fetch them
to the environment. To avoid this
monolithic for BT app is disabled.

Signed-off-by: Yassine El Aissaoui <yassine.elaissaoui@nxp.com>

dleach02 commented on tests/bluetooth/tester/testcase.yaml:
```diff
@@ -6,9 +6,16 @@ tests:
     tags: bluetooth
     harness: bluetooth
+  bluetooth.general.tester.nxp:
+    # Disabling monolithic since CI environment doesn't use blobs
```
is it the "build_only" setting that does this?

dleach02 replied:
nevermind... I see the disabling of the monolithic kconfig setting

decsny commented:
why do you have the wording as linked "dynamically", what is dynamic about it? "dynamic linking" has a [specific meaning](https://en.wikipedia.org/wiki/Dynamic_linker), but I don't think this is doing dynamic linking, am I misunderstanding something?

yeaissa committed:
boards: rd_rw612_bga: Update board documentation

Add information about BLE.

Signed-off-by: Yassine El Aissaoui <yassine.elaissaoui@nxp.com>

yeaissa commented:
@decsny, this feature allows to link the binary blob with the application which forms a single monolithic image that can be flashed directly to the board. Without it, the user has to flash the binary blob manually since it won't be linked to the application image.
We have added a doc section for BLE on the latest commit, please give it a look.

decsny commented:
> @decsny, this feature allows to link the binary blob with the application which forms a single monolithic image that can be flashed directly to the board. Without it, the user has to flash the binary blob manually since it won't be linked to the application image.
> We have added a doc section for BLE on the latest commit, please give it a look. 

I understand what the PR is doing, what I'm saying is in your commit messages you are saying "dynamic linking" which I don't think is what you are really doing

yeaissa commented:
> > @decsny, this feature allows to link the binary blob with the application which forms a single monolithic image that can be flashed directly to the board. Without it, the user has to flash the binary blob manually since it won't be linked to the application image.
> > We have added a doc section for BLE on the latest commit, please give it a look.
> 
> I understand what the PR is doing, what I'm saying is in your commit messages you are saying "dynamic linking" which I don't think is what you are really doing

@decsny you are right. Updated the commit messages, and removed the misleading "dynamic linking".

kartben commented on boards/nxp/rd_rw612_bga/doc/index.rst:
```diff
@@ -134,6 +146,26 @@ should see the following message in the terminal:
+BLE functionality requires to fetch binary blobs, so make sure to follow
+the ``Fetch Binary Blobs`` section first.
+
+Those binary blobs can be used in two different ways, depending if ``CONFIG_NXP_MONOLITHIC_BT``
```
```suggestion
Those binary blobs can be used in two different ways, depending if :kconfig:option:`CONFIG_NXP_MONOLITHIC_BT`
```

kartben commented on boards/nxp/rd_rw612_bga/doc/index.rst:
```diff
@@ -134,6 +146,26 @@ should see the following message in the terminal:
+Those binary blobs can be used in two different ways, depending if ``CONFIG_NXP_MONOLITHIC_BT``
+is enabled or not:
+
+- ``CONFIG_NXP_MONOLITHIC_BT`` is enabled (default):
```
```suggestion
- :kconfig:option:`CONFIG_NXP_MONOLITHIC_BT` is enabled (default):
```

kartben commented on boards/nxp/rd_rw612_bga/doc/index.rst:
```diff
@@ -134,6 +146,26 @@ should see the following message in the terminal:
+one single monolithic image.
+The user has nothing else to do other than flashing the application to the board.
+
+- ``CONFIG_NXP_MONOLITHIC_BT`` is disabled:
```
```suggestion
- :kconfig:option:`CONFIG_NXP_MONOLITHIC_BT` is disabled:
```

yeaissa commented:
Updated, Thanks @kartben for the suggestions.

yeaissa commented:
@jhedberg, Could you please review this PR ?

yeaissa commented:
Updated west sha.

dleach02 commented:
@jhedberg we merged the HAL change which purged the approvals...
